### PR TITLE
[Distributed] Fix generic actors and distributed funcs returning those generics

### PR DESCRIPTION
**Pull request by ktoso**
Depends on https://github.com/apple/swift/pull/41616

The actual fix: https://github.com/apple/swift/pull/41747/commits/5c54ad2f2a8e9bfb97a5bb31f4c274bd14cef88e

~Once we merge https://github.com/apple/swift/pull/41616 I'll rebase this.~ Done

Resolves rdar://84458365

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 1]
```diff
@@ -523,10 +525,10 @@ static FuncDecl *createDistributedThunkFunction(FuncDecl *func) {
       paramName = C.getIdentifier("p" + llvm::utostr(i));
     }
 
-    auto paramDecl = new (C)
-        ParamDecl(SourceLoc(),
-                  /*argumentNameLoc=*/SourceLoc(), funcParam->getArgumentName(),
-                  /*parameterNameLoc=*/SourceLoc(), paramName, DC);
+    auto paramDecl = new (C) ParamDecl(SourceLoc(),
+                               SourceLoc(), funcParam->getArgumentName(),
+                               SourceLoc(), funcParam->getParameterName(),
```
Comment (xedin) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:530]: This has to use `paramName`, the parameter cannot be anonymous if it's used in the body.

Comment (ktoso) [reply lib/Sema/CodeSynthesisDistributedActor.cpp:530]: Oh, thanks for catching that

Comment (ktoso) [reply lib/Sema/CodeSynthesisDistributedActor.cpp:530]: I messed up the rebase picking the wrong impl here it seems, thanks

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
This actually still fails in SIL handling in the init/deinit when the ACTOR is generic. I'll handle this separately.

**Comment by ktoso** [comment]
Okey this actually still has missing symbols that I noticed that I added runtime tests... I'll give it a sleep tonight but will try to make progress tomorrow

**Comment by ktoso** [comment]
Note to self: this one over to @DougGregor to see if he can figure it out

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test Linux

**Comment by ktoso** [comment]
Thank you a lot @xedin!  🙇 Looking good, I cleaned up a bit and added a minor test just to be sure remote call was fine as well (it was).

@swift-ci please smoke test and merge

File: lib/SIL/IR/SILFunction.cpp [hunk 2]
```diff
@@ -319,6 +319,8 @@ SILType SILFunction::mapTypeIntoContext(SILType type) const {
 
 SILType GenericEnvironment::mapTypeIntoContext(SILModule &M,
                                                SILType type) const {
+  if (type.hasArchetype())
+    type.dump();
```
Comment (xedin) [inline lib/SIL/IR/SILFunction.cpp:323]: You forgot to remove `dump` :)

Comment (ktoso) [reply lib/SIL/IR/SILFunction.cpp:323]: Oh no 🙈

Comment (ktoso) [reply lib/SIL/IR/SILFunction.cpp:323]: Fixed and had to rebase anyway.

**Comment by ktoso** [comment]
@swift-ci please smoke test and merge

**Pull request merged**